commands/box/transaction/root.go

```diff
@@ -204,61 +228,165 @@ func sendrawtx(cmd *cobra.Command, args []string) {
 		return
 	}
-	hash := crypto.HashType{}
-	hash.SetString(args[0])
-	conn, err := rpcutil.GetGRPCConn(getRPCAddr())
+	respRPC, err := rpcutil.RPCCall(rpcpb.NewTransactionCommandClient, "GetRawTransaction",
+		&rpcpb.GetRawTransactionRequest{Hash: args[0]}, common.GetRPCAddr())
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
-	defer conn.Close()
-	tx, err := rpcutil.GetRawTransaction(conn, hash.GetBytes())
+	resp, ok := respRPC.(*rpcpb.GetRawTransactionResponse)
+	if !ok {
+		fmt.Println("Convertion to rpcpb.GetRawTransactionResponse failed")
+		return
+	}
+	if resp.Code != 0 {
+		fmt.Println(resp.Message)
+		return
+	}
+	tx := new(types.Transaction)
+	err = tx.FromProtoMessage(resp.Tx)
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
 	fmt.Println(format.PrettyPrint(tx))
+
 }
 
-func getBlockDetailCmdFunc(cmd *cobra.Command, args []string) {
-	fmt.Println("getBlockDetail called")
-	if len(args) < 1 {
-		fmt.Println("Param txhash required")
+func getTxCountByHash(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		fmt.Println(cmd.Use)
 		return
 	}
-	hash := args[0]
-	conn, err := rpcutil.GetGRPCConn(getRPCAddr())
+	respRPC, err := rpcutil.RPCCall(rpcpb.NewContorlCommandClient, "GetBlockTransactionCountByHash",
+		&rpcpb.GetBlockTransactionCountByHashReq{BlockHash: args[0]}, common.GetRPCAddr())
 	if err != nil {
 		fmt.Println(err)
 		return
 	}
-	defer conn.Close()
-	//
-	block, err := rpcutil.GetBlock(conn, hash)
+	resp, ok := respRPC.(*rpcpb.GetBlockTxCountResp)
+	if !ok {
+		fmt.Println("Convertion to rpcpb.GetBlockTxCountResp failed")
+		return
+	}
+	if resp.Code != 0 {
+		fmt.Println(resp.Message)
+		return
+	}
+	fmt.Println("transaction amount:", resp.Count)
+}
+
+func getTxCountByHeight(cmd *cobra.Command, args []string) {
+	if len(args) != 1 {
+		fmt.Println(cmd.Use)
+		return
+	}
+	height, err := strconv.ParseUint(args[0], 10, 64)
+	if err != nil {
+		fmt.Println("the type of height conversion failed:", err)
+		return
+	}
+	respRPC, err := rpcutil.RPCCall(rpcpb.NewContorlCommandClient, "GetBlockTransactionCountByHeight",
+		&rpcpb.GetBlockTransactionCountByHeightReq{Height: uint32(height)}, common.GetRPCAddr())
+	if err != nil {
+		fmt.Println("RPC call failed:", err)
+		return
+	}
+	resp, ok := respRPC.(*rpcpb.GetBlockTxCountResp)
+	if !ok {
+		fmt.Println("Conversion to rpcpb.GetBlockTxCountResp failed")
+		return
+	}
+	if resp.Code != 0 {
+		fmt.Println(resp.Message)
+		return
+	}
+	fmt.Println("transaction amount:", resp.Count)
+}
+
+func getTxByHash(cmd *cobra.Command, args []string) {
+	if len(args) != 2 {
+		fmt.Println(cmd.Use)
+		return
+	}
+	index, err := strconv.ParseUint(args[1], 10, 64)
+	if err != nil {
+		fmt.Println("the type of index conversion failed:", err)
+		return
+	}
+	respRPC, err := rpcutil.RPCCall(rpcpb.NewContorlCommandClient, "GetTransactionByBlockHashAndIndexReq",
+		&rpcpb.GetTransactionByBlockHashAndIndexReq{BlockHash: args[0], Index: uint32(index)}, common.GetRPCAddr())
```

Comment:
function name is getTxByHash, why the request includes block hash and index
on the other way, only need to use ViewTxDetail api

Reply:
no change?